test/shardingTableStorage.test.ts

```diff
@@ -0,0 +1,213 @@
+    const headValue = await ShardingTableStorage.head();
+
+    expect(headValue.toNumber()).to.equal(2);
+  });
+
+  it('Set profile identityId to be tail, expect to pass', async () => {
+    await ShardingTableStorage.setTail(2);
+    const tailValue = await ShardingTableStorage.tail();
+
+    expect(tailValue.toNumber()).to.equal(2);
+  });
+
+  it('Check node existence with valid identity id, expect to pass', async () => {
+    identityId = await createProfile();
+    await ShardingTableStorage.createNodeObject(identityId, identityId, identityId);
+    const nodeExists = await ShardingTableStorage.nodeExists(identityId);
+
+    expect(nodeExists).to.be.true;
+  });
+
+  it('Set new previous identity id, expect to pass', async () => {
+    identityId = await createProfile();
+    const newPrevIdentityId = 2;
+
+    await ShardingTableStorage.createNodeObject(identityId, identityId, identityId);
+    await ShardingTableStorage.setPrevIdentityId(identityId, newPrevIdentityId);
+    const nodeResult = await ShardingTableStorage.getNode(identityId);
+
+    expect(nodeResult.identityId.toNumber()).to.equal(identityId);
+    expect(nodeResult.prevIdentityId.toNumber()).to.equal(newPrevIdentityId);
+    expect(nodeResult.nextIdentityId.toNumber()).to.equal(identityId);
+  });
+
+  it('Set new next identity id, expect to pass', async () => {
+    identityId = await createProfile();
+    const newNextIdentityId = 3;
+
+    await ShardingTableStorage.createNodeObject(identityId, identityId, identityId);
+    await ShardingTableStorage.setNextIdentityId(identityId, newNextIdentityId);
+    const nodeResult = await ShardingTableStorage.getNode(identityId);
+
+    expect(nodeResult.identityId.toNumber()).to.equal(identityId);
+    expect(nodeResult.prevIdentityId.toNumber()).to.equal(identityId);
+    expect(nodeResult.nextIdentityId.toNumber()).to.equal(newNextIdentityId);
+  });
+
+  it('Create 4 nodes, node 4 linked with node 2 and 3, get multiple nodes, expect to pass', async () => {
+    // create 3 nodes and set node object for prev and next identityId
+    const profiles = await createMultipleProfiles();
+    await ShardingTableStorage.createNodeObject(profiles[1], profiles[0], profiles[2]); // identityId = 2, prevIdentityId = 1, nextIdentityId = 3
+    const getNodeResult = await ShardingTableStorage.getNode(profiles[1]);
+
+    expect(getNodeResult.identityId.toNumber()).to.equal(profiles[1]);
+    expect(getNodeResult.prevIdentityId.toNumber()).to.equal(profiles[0]);
+    expect(getNodeResult.nextIdentityId.toNumber()).to.equal(profiles[2]);
+
+    // add 1 more node
+    const adminWallet3 = await Profile.connect(accounts[3]);
+    const newProfile = await adminWallet3.createProfile(accounts[0].address, nodeId4, 'Token4', 'TKN4');
+
+    const receipt1 = await newProfile.wait();
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    const newIdentityId = receipt1.events[3].args.identityId.toNumber();
```

Comment:
Not sure about this, but I guess it should work without toNumber(), using BigNumber object